src/OpenTelemetry.Instrumentation.AspNetCore/AspNetCoreInstrumentation.cs

```diff
@@ -23,11 +24,23 @@ namespace OpenTelemetry.Instrumentation.AspNetCore
     /// </summary>
     internal class AspNetCoreInstrumentation : IDisposable
     {
+        private static readonly HashSet<string> DiagnosticSourceEvents = new()
+        {
+            "Microsoft.AspNetCore.Hosting.HttpRequestIn.Start",
+            "Microsoft.AspNetCore.Hosting.HttpRequestIn.Stop",
+            "Microsoft.AspNetCore.Mvc.BeforeAction",
+            "Microsoft.AspNetCore.Diagnostics.UnhandledException",
+            "Microsoft.AspNetCore.Hosting.UnhandledException",
+        };
+
+        private readonly Func<string, object, object, bool> isEnabled = (eventName, obj1, obj2)
+            => DiagnosticSourceEvents.Contains(eventName);
+
         private readonly DiagnosticSourceSubscriber diagnosticSourceSubscriber;
 
         public AspNetCoreInstrumentation(HttpInListener httpInListener)
         {
-            this.diagnosticSourceSubscriber = new DiagnosticSourceSubscriber(httpInListener, null);
+            this.diagnosticSourceSubscriber = new DiagnosticSourceSubscriber(httpInListener, this.isEnabled);
```

Comment:
👍 I think using an `isEnabled` delegate to define the events we're interested in is a good step in the right direction.

If we were to do this to all our instrumentation then in the end I think we'd still want to do away with pattern matching here https://github.com/open-telemetry/opentelemetry-dotnet/blob/27658f058cc248500805d66ae2ab65c2099d6400/src/OpenTelemetry/DiagnosticSourceInstrumentation/DiagnosticSourceListener.cs#L43.

That is, I no longer see the value this really brings. I think it makes more sense to change [ListenerHandler](https://github.com/open-telemetry/opentelemetry-dotnet/blob/27658f058cc248500805d66ae2ab65c2099d6400/src/OpenTelemetry/DiagnosticSourceInstrumentation/ListenerHandler.cs#L23) to just have one method like the following and have it switch on event name - much like `OnCustom` does today.

```csharp
public virtual void OnEventWritten(Activity activity, string eventName, object payload)
```

All that said, I think we could do this in stages. So leave `ListenerHandler` alone for now. Do what you've done here to define the events we want. Then come back later and decide whether we want to refactor `ListenerHander` and friends.

Reply:
good that all these DS helpers are kept internal. we can break them apart.

Reply:
even Activity is not required to be part of this OnEventWritten. It gives wrong information that Activity is somehow part of the event. It is not. It is part of ambient context, and anyone interested can use Activity.Current to get it. Or use other mechanism (like from inside httpcontext somewhere etc) if Activity.Current is not reliable.

Reply:
I will do a follow up PR for moving all events to custom callback.